meter/tesla.go

```diff
@@ -101,10 +130,38 @@ func NewTesla(uri, usage string) (api.Meter, error) {
 	return decorateTesla(m, totalEnergy, batterySoC), nil
 }
 
+// Login calls login and saves the returned cookie
+func (m *Tesla) Login() error {
+	// username for the powerwall seems to always be customer; email is not required for authentication
+	payload := map[string]interface{}{"password": m.password, "username": "customer"}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	// returns cookie which is saved in the cookie jar
+	resp, err := m.Client.Post(m.uri.String()+teslaLoginURI, "application/json", bytes.NewReader(payloadBytes))
+	if err != nil {
+		return fmt.Errorf("failed to post: %s", err)
+	}
+
+	defer resp.Body.Close()
+	// failed to login
+	if resp.StatusCode != http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
please use the go 1.16 io package, ioutil is deprecated

Reply:
Might also be possible to use `request.DoBody` and ignore the returned body, might be less lines of code.

Reply:
I think it is useful to propagate the error here. The Powerwall sometimes does weird stuff and in such cases it is often helpful to have access to the exact errors. But I found a way to reduce the lines of code.